cdc/redo/writer/memory/mem_log_writer_test.go

```diff
@@ -106,12 +106,30 @@ func testWriteEvents(t *testing.T, events []writer.RedoEvent) {
 
 	functions := map[string]func(error){
 		"WriteEvents": func(expected error) {
-			err := lw.WriteEvents(ctx, events...)
-			require.ErrorIs(t, errors.Cause(err), expected)
+			if expected == nil {
+				err := lw.WriteEvents(ctx, events...)
+				require.Nil(t, err)
```

Comment:
require.NoError is prefered.

Also , I am curious, why the expected may be nil and not nil at the same case?

Reply:
Only the first call raises an error and It is uncertain which function is called